WooCommerce/Classes/ViewRelated/Blaze/AdDestination/BlazeAdDestinationSettingViewModel.swift

```diff
@@ -66,20 +71,48 @@ final class BlazeAdDestinationSettingViewModel: ObservableObject {
         )
     }
 
+    var shouldDisableAddParameterButton: Bool {
+        calculateRemainingCharacters() == 0
+    }
+
+    private let onSave: (String) -> Void
+
     init (productURL: String,
           homeURL: String,
-          selectedDestinationType: DestinationURLType = .product,
-          parameters: [BlazeAdURLParameter] = []) {
+          finalDestinationURL: String,
+          onSave: @escaping (String) -> Void) {
         self.productURL = productURL
         self.homeURL = homeURL
-        self.selectedDestinationType = selectedDestinationType
-        self.parameters = parameters
+        self.initialFinalDestinationURL = finalDestinationURL
+        self.onSave = onSave
+
+        setStartingDestinationType()
+        setStartingParameters()
+    }
+
+    private func setStartingDestinationType() {
+        if initialFinalDestinationURL.hasPrefix(productURL) {
+            selectedDestinationType = .product
+        } else {
+            selectedDestinationType = .home
+        }
+    }
+
+    private func setStartingParameters() {
```

Comment:
Opinion: What do you think about renaming these methods into `initializeParameters` or `initParameters`? Same for the other `setStartingDestinationType` method. IMO, `starting` might be confusing.

Reply:
I agree! Renamed as part of https://github.com/woocommerce/woocommerce-ios/pull/11703/commits/206937a67ac42e68eb2a3ec052fe35b84ba08030